pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -176,6 +176,15 @@ func (c *Cluster) Start(clusterInfo *cephconfig.ClusterInfo, rookVersion string,
 		return nil, fmt.Errorf("failed to initialize ceph cluster info. %+v", err)
 	}
 
+	availableNodes, _, err := c.getAvailableMonNodes()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get available mon nodes. %+v", err)
+	}
+	nodeCount := len(availableNodes)
+	if !c.spec.Mon.AllowMultiplePerNode && c.spec.Mon.Count > nodeCount {
+		return nil, fmt.Errorf("the desired mon count is %d while the number of available nodes is %d despite allowMultiplePerNode is %v", c.spec.Mon.Count, nodeCount, c.spec.Mon.AllowMultiplePerNode)
```

Comment:
This looks like the correct place for an error, just a minor suggestion: `%t` can be used to print a bool

Reply:
Thanks for the tips!